prerender-until-script.md

```diff
@@ -0,0 +1,123 @@
+*   A `scripting mode` is added to the `navigable` struct, with a state of `paused-until-activation`.
+*   When a page is prerendered with `prerender_until_script`, all script execution is paused. This means that no `<script>` tags are executed, and no timers or promises are allowed to execute their callbacks. The JavaScript engine is effectively "frozen" until the document is activated. Upon activation, script execution resumes as if there was no interruption.
+*   Upon activation, the `finalize activation` algorithm sets the `scripting mode` to `enabled` and resumes script execution.
+
+## Developer Impact
+
+For developers, this is a powerful new tool for performance optimization. Pages that were previously poor candidates for prerendering due to their reliance on JavaScript can now benefit from this feature.
+
+## Security and Privacy Considerations
+
+The `prerender_until_script` action inherits all the security and privacy protections of the standard prerendering mechanism. All restrictions, such as those on cross-site navigations and access to user-facing APIs, remain in effect.
+
+This feature does not introduce any new security or privacy concerns beyond those already addressed by the prerendering specification. By deferring script execution until activation, it offers a model with a more limited execution context during the speculation phase, which can be a desirable characteristic for some applications.
+
+## Considered Alternatives
+
+### No Prerendering
+
+The simplest alternative is to not prerender pages with heavy scripts. However, this means that users navigating to these pages will not experience the performance benefits of prerendering. `prerender_until_script` provides a better option by allowing the most time-consuming parts of the page load (network and parsing) to be done in advance.
+
+### Modifying Pages to be Prerender-Friendly
+
+Another alternative is for developers to modify their pages to be more "prerender-friendly," for example, by delaying script execution manually using the `prerenderingchange` event. While this is a valid approach, `prerender_until_script` provides a simpler, declarative way to achieve a similar result without requiring significant code changes. It also provides a stronger guarantee that *no* scripts will be executed, which can be difficult to achieve manually.
+
+### Alternative Naming
+
+The initial proposal for this feature used the name `preparse`. Other names, such as `prescan` and `prefetch_with_subresources`, were also considered. The name `prerender_until_script` was chosen to make it clear that this is a form of prerendering, but with a specific modification to the script execution lifecycle. This naming helps to place the feature within the existing mental model of the Speculation Rules API.
+
+## Stakeholder Feedback
+
+*This section is a placeholder for feedback from browser vendors, web developers, and other stakeholders.*
```

Comment:
Move the link to #305 into this section and briefly that we have at least two developers who are very positive on this.

Maybe also just have a general mention on how many times we've heard about people being unable to go beyond prefetch, all the way to prerender.

Reply:
Thank you!